futures-executor/src/enter.rs

```diff
@@ -21,12 +21,14 @@ pub struct EnterError {
 /// Marks the current thread as being within the dynamic extent of an
 /// executor.
 ///
-/// Executor implementations should call this function before blocking the
-/// thread.
+/// Executor implementations should call this function before beginning to
+/// execute any tasks. Doing so ensures that executors aren't accidentally
+/// invoked in a nested fashion.
 ///
 /// # Error
```

Comment:
It would be good to include a comment about keeping the `Enter` result around until you're *done* blocking the task, and not holding onto it after you've finished blocking.